### PR TITLE
[#56609] Fix further possible TypeErrors for gitlab hooks

### PR DESCRIPTION
**Pull request by judithroth**
See https://community.openproject.org/projects/openproject/work_packages/56609
And https://appsignal.com/openproject-gmbh/sites/62b06dacd2a5e41321946fcf/exceptions/incidents/699

### CODE REVIEW

File: modules/gitlab_integration/spec/lib/open_project/gitlab_integration/notification_handler/push_hook_spec.rb
Comment (mereghost) [inline modules/gitlab_integration/spec/lib/open_project/gitlab_integration/notification_handler/push_hook_spec.rb:141]: 🟡 While the test mentions not raising, there's no assertion to it. I get why but I'd prefer for it either have one assertion like `expect { process }.to_not raise_error(ExceptionOfInterest)` and the message expectation or the description changed to something along the lines of `handles commits without messages`

**Review by mereghost — APPROVED** [review]
Great work here.

I've used an traffic light system to make notes on what I think is:
🟢 Just me being weird or minor suggestions
🟡 Points that might warrant clarification discussion
🔴 (which I use sparingly) Major issues that need to be addressed before a merge

File: modules/gitlab_integration/spec/lib/open_project/gitlab_integration/notification_handler/push_hook_spec.rb
Comment (judithroth) [reply modules/gitlab_integration/spec/lib/open_project/gitlab_integration/notification_handler/push_hook_spec.rb:141]: Good point. When writing my thinking was that the test would raise and therefore "fail" if there still was an error, but I also agree that making things explicit is often the better choice.
So I added your suggested expectation, but since you already approved I assume you don't want to review again, right? :slightly_smiling_face:

Comment (mereghost) [reply modules/gitlab_integration/spec/lib/open_project/gitlab_integration/notification_handler/push_hook_spec.rb:141]: That was my only point and wasn't something that I feel should even block the merge, so feel free to :shipit: =)

**Pull request merged**
